### PR TITLE
ETL SubProject

### PR DESCRIPTION
**Pull request by echeipesh**
This PR replaces the commands in `geotrellis.spark.ingest` with a subproject that makes easy to write Ingest/ETL like so:

```scala
object GeoTrellisETL extends App {
  val etl = Etl[SpatialKey](args, S3Module, HadoopModule)
  implicit val sc = SparkUtils.createSparkContext("GeoTrellis ETL")
  val (id, rdd) = etl.ingest()
  val result = rdd.localAdd(1)
  etl.save(id, result, ZCurveKeyIndexMethod)
  sc.stop()
}
```

The choice and configuration of ingest/output modules are delayed until run-time though plugin architecture implemented with Guice.

### CODE REVIEW

**Comment by jwalgran** [comment]
It would be helpful to update ``publish-local.sh`` to include the new ``spark-etl`` project introduced in this PR

File: spark-etl/src/main/scala/geotrellis/spark/etl/EtlConf.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package geotrellis.spark.etl
+
+import geotrellis.proj4.CRS
+import org.apache.spark.storage.StorageLevel
+import org.rogach.scallop._
+
+class EtlConf(args: Seq[String]) extends ScallopConf(args){
+  val ingest       = opt[String]("ingest", required = true,
```
Comment (lossyrob) [inline spark-etl/src/main/scala/geotrellis/spark/etl/EtlConf.scala]: rename `ingest` here to `input`, and then make sure it's consistent through codebase.

File: spark-etl/src/main/scala/geotrellis/spark/etl/IngestPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import geotrellis.proj4.CRS
+import geotrellis.spark.RasterRDD
+import geotrellis.spark.tiling.{LayoutLevel, LayoutScheme}
+import org.apache.spark.SparkContext
+import org.apache.spark.storage.StorageLevel
+
+import scala.reflect._
+
+trait IngestPlugin {
```
Comment (lossyrob) [inline spark-etl/src/main/scala/geotrellis/spark/etl/IngestPlugin.scala]: `EtlInputPlugin`

File: spark-etl/src/main/scala/geotrellis/spark/etl/SinkPlugin.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package geotrellis.spark.etl
+
+import geotrellis.spark.io.index.KeyIndexMethod
+import geotrellis.spark.{LayerId, RasterRDD}
+import scala.reflect.ClassTag
+
+trait SinkPlugin {
```
Comment (lossyrob) [inline spark-etl/src/main/scala/geotrellis/spark/etl/SinkPlugin.scala]: `OutputPlugin`

File: spark/ingest-nlcd-s3.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+export JAR=target/scala-2.10/geotrellis-spark-assembly-0.10.0-SNAPSHOT.jar
```
Comment (lossyrob) [inline spark/ingest-nlcd-s3.sh]: rm

File: spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    (ingest, output)
+  }
+
+  def ingest()(implicit sc: SparkContext): (LayerId, RasterRDD[K]) = {
+    val (level, rdd) = ingestPlugin[K](conf.cache(), conf.crs(), ZoomedLayoutScheme(conf.tileSize()), conf.ingestProps)
+    LayerId(conf.layerName(), level.zoom) -> rdd
+  }
+
+  def save(id: LayerId, rdd: RasterRDD[K], method: KeyIndexMethod[K]): Unit = {
```
Comment (echeipesh) [inline spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala]: @lossyrob does this become "output" ?
I'd like to suggest naming load/save to keep functions as verbs vs nouns.

**Pull request merged**